File: src/main/java/com/wikinode/studio/model/SourceOperation.java
package com.wikinode.studio.model;

public record SourceOperation(
  String operationId,
  String operationType,
  String sourceId,
  String knowledgeBaseId,
  String rawMaterialId,
  String parsedDocumentId,
  String status,
  String requestedBy,
  String startedAt,
  String finishedAt,
  String summary,
  String errorSummary
) {
  public SourceOperation(
    String operationId,
    String operationType,
    String sourceId,
    String rawMaterialId,
    String parsedDocumentId,
    String status,
    String requestedBy,
    String startedAt,
    String finishedAt,
    String summary,
    String errorSummary
  ) {
    this(
      operationId,
      operationType,
      sourceId,
      null,
      rawMaterialId,
      parsedDocumentId,
      status,
      requestedBy,
      startedAt,
      finishedAt,
      summary,
      errorSummary
    );
  }
}

File: src/types/source-operation.ts
export type SourceOperation = {
  operationId: string
  operationType: string
  sourceId: string
  rawMaterialId?: string | null
  parsedDocumentId?: string | null
  status: string
  requestedBy: string
  startedAt: string
  finishedAt?: string | null
  summary: string
  errorSummary?: string | null
}

export type SourceIngestionRunRequest = {
  conversionProfile?: string
  requestedBy?: string
}

export type SourceIngestionRunResult = {
  operationId: string
  sourceId: string
  status: "succeeded" | "skipped" | "failed"
  summary: string
  rawMaterialCount: number
  parsedDocumentCount: number
  generatedSuggestionIds: string[]
  skippedParsedDocumentIds: string[]
}

export type SourceImportResult = {
  operationId: string
  sourceId: string
  rawMaterialId: string
  parsedDocumentId: string
  status: "succeeded" | "skipped" | "failed"
  summary: string
  segmentCount: number
  segmentIds: string[]
  suggestionId?: string | null
}
